src/Storages/ObjectStorage/DataLakes/IStorageDataLake.h

```diff
@@ -102,12 +115,28 @@ class IStorageDataLake final : public StorageObjectStorage
         Storage::updateConfiguration(local_context);
 
         auto new_metadata = DataLakeMetadata::create(Storage::object_storage, base_configuration, local_context);
-        if (current_metadata && *current_metadata == *new_metadata)
-            return;
```

Comment:
Now when we removed this change, we do extra work in `current_metadata->getDataFileInfos()` every time even when there is not changes in metadata